velox/functions/prestosql/window/tests/RowNumberTest.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/dwio/common/tests/utils/BatchMaker.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/window/WindowFunctionsRegistration.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+class RowNumberTest : public OperatorTestBase {
+ protected:
+  void SetUp() {
+    exec::test::OperatorTestBase::SetUp();
+    velox::window::registerWindowFunctions();
+  }
+
+  std::vector<RowVectorPtr> makeVectors(
+      const std::shared_ptr<const RowType>& rowType,
+      vector_size_t size,
+      int numVectors,
+      std::function<bool(vector_size_t /*index*/)> isNullAt) {
+    std::vector<RowVectorPtr> vectors;
+    for (int32_t i = 0; i < numVectors; ++i) {
+      auto vector = std::dynamic_pointer_cast<RowVector>(
+          velox::test::BatchMaker::createBatch(
+              rowType, size, *pool_, isNullAt));
+      vectors.push_back(vector);
+    }
+    return vectors;
+  }
+
+  std::shared_ptr<const RowType> rowType_{
+      ROW({"c0", "c1", "c2", "c3", "c4", "c5", "c6"},
+          {BIGINT(),
+           SMALLINT(),
+           INTEGER(),
+           BIGINT(),
+           REAL(),
+           DOUBLE(),
+           VARCHAR()})};
+  folly::Random::DefaultGenerator rng_;
+};
+
+TEST_F(RowNumberTest, basicRowNumber) {
+  auto testWindowSql = [&](const RowVectorPtr& input,
+                           const std::string& windowSql) -> void {
+    VELOX_CHECK_GE(input->size(), 2);
+
+    auto op = PlanBuilder()
+                  .values({input})
+                  .project({"c0 as c0", "c1 as c1"})
+                  .window({windowSql})
+                  .orderBy({"c0 asc nulls last", "c1 asc nulls last"}, false)
+                  .planNode();
+    assertQuery(
+        op, "SELECT c0, c1, " + windowSql + " FROM tmp order by c0, c1");
```

Comment:
order by -> ORDER BY

If you want to ensure the order of the output, use assertQuery method that takes sortingKeys.